common-lib/src/main/scala/com/gu/mediaservice/GridClient.scala

```diff
@@ -132,73 +94,91 @@ class GridClient(apiKey: String, services: Services, maxIdleConnections: Int, de
+                                   url: URL
+                                 ): Response = {
+//    val res = processResponse(response, url)
+    response.status match {
+      case 200 => Found(Json.parse(response.body), response)
+      case 404 => NotFound(response.body, response)
+      case failCode => Error(failCode, url, response)
+    }
+  }
 
-      val errorJson = Json.obj(
-        "level" -> "ERROR",
-        "errorStatusCode" -> statusCode,
-        "message" -> Json.obj(
-          "errorMessage" -> errorMessage,
-          "downstreamErrorMessage" -> downstreamErrorMessage
-        )
-      )
-      logger.error(errorJson.toString())
-      throw new DownstreamApiInBadStateException(errorMessage, downstreamErrorMessage)
+  def validateApiKey(projectionEndpoint: String, authFn: WSRequest => WSRequest)
+                    (implicit ec: ExecutionContext): Future[Boolean] = {
+    val projectionUrl = new URL(s"$projectionEndpoint/")
+    for {
+      response <- makeGetRequestAsync(projectionUrl, authFn)
+    } yield response match {
+      case Found(_, _) => true
+      case NotFound(_, _) => true
+      case Error(_, _, _) => throw new Exception("Authorisation failed")
     }
   }
 
-  def getImageLoaderProjection(mediaId: String, imageLoaderEndpoint: String): Option[Image] = {
+  def getImageLoaderProjection(mediaId: String, imageLoaderEndpoint: String, authFn: WSRequest => WSRequest)
+                              (implicit ec: ExecutionContext): Future[Option[Image]] = {
     logger.info("attempt to get image projection from image-loader")
-    val url = new URL(s"${imageLoaderEndpoint}/images/project/$mediaId")
-    val res = makeGetRequestSync(url, apiKey)
-    validateResponse(res, url)
-    logger.info(s"got image projection from image-loader for $mediaId with status code $res.statusCode")
-    if (res.statusCode == 200) Some(res.body.as[Image]) else None
+    val url = new URL(s"$imageLoaderEndpoint/images/project/$mediaId")
+    for {
+      response <- makeGetRequestAsync(url, authFn)
+    } yield response match {
+      case Found(json, _) => Some(json.as[Image])
+      case NotFound(_, _) => None
+      case e@Error(_, _, _) => e.logError
+    }
   }
 
-  def getLeases(mediaId: String)(implicit ec: ExecutionContext): Future[LeasesByMedia] = {
+  def getLeases(mediaId: String, authFn: WSRequest => WSRequest)(implicit ec: ExecutionContext): Future[LeasesByMedia] = {
     logger.info("attempt to get leases")
     val url = new URL(s"${services.leasesBaseUri}/leases/media/$mediaId")
-    makeGetRequestAsync(url, apiKey).map { res =>
-      validateResponse(res, url)
-      if (res.statusCode == 200) (res.body \ "data").as[LeasesByMedia] else LeasesByMedia.empty
+    for {
+      response <- makeGetRequestAsync(url, authFn)
+    } yield response match {
+      case Found(json, _) => (json \ "data").as[LeasesByMedia]
+      case NotFound(_, _) => LeasesByMedia.empty
+      case e@Error(_, _, _) => e.logError
     }
```

Comment:
This is identical to the code below if you don't like for comprehensions (and I don't think it adds anything here).

```suggestion
   makeGetRequestAsync(url, authFn).map {
      case Found(json, _) => (json \ "data").as[LeasesByMedia]
      case NotFound(_, _) => LeasesByMedia.empty
      case e@Error(_, _, _) => e.logError
    }
```

Reply:
Done.